Projects/App/Sources/Presentation/Room/PlaceVote/PlaceVoteCell.swift

```diff
@@ -0,0 +1,233 @@
+    
+    
+    // MARK: - Initializers
+    
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        setUI(progress: 1.0)
+        setStatus(type: .doneNotSelected)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func setSelected(_ selected: Bool, animated: Bool) {
+       super.setSelected(selected, animated: animated)
+        self.onSelected()
+    }
+    
+}
+
+
+// MARK: privates
+extension PlaceVoteCell {
+    
+    func configure(with text: String) {
+        nameLabel.text = "\(text)번째"
+    }
+    
+    func onSelected() {
+        isSlected = !isSlected
+        manageSelection()
+        self.layoutIfNeeded()
+    }
+    
+    private func setStatus(type: PlaceVoteCellType) {
+        switch type {
+        case .progSelected:
+            containerView.layer.borderWidth = 1.0
+            containerView.layer.borderColor = UIColor.wwwColor(.WWWGreen).cgColor
+            badgeView.backgroundColor = UIColor.wwwColor(.WWWGreen)
+            makeGradient(opacity: 0.5)
+            showBadge()
+        case .progNotSelected:
+            containerView.layer.borderWidth = 0
+            hideGradient()
+            hideBadge()
+        case .doneSelected:
+            badgeView.backgroundColor = .clear
+            makeGradient(opacity:1.0)
+            showBadge()
+        case .doneNotSelected:
+            makeGradient(opacity: 0.1)
+            hideBadge()
+        }
+    }
+
+}
+
+
+
+// MARK: - Private Methods
+extension PlaceVoteCell {
+    
+    private func setUI(progress: CGFloat) {
+        contentView.addSubviews(containerView, colorView)
+
+        containerView.snp.makeConstraints { make in
+            make.edges.equalToSuperview().inset(UIEdgeInsets(top: 6, left: 20, bottom: 6, right: 20))
+            make.height.equalTo(48)
+        }
+        
+        containerView.addSubview(colorView)
+        colorView.layer.cornerRadius = 15
+        colorView.layer.masksToBounds = true
+        hideGradient()
+        colorView.snp.makeConstraints { make in
+            make.leading.top.equalToSuperview()
+            make.height.equalTo(48)
+        }
+        
+        let width = (self.containerView.bounds.width) * progress // TODO: 계산이상하게 안맞음
+        containerView.addSubview(colorView)
+        colorView.snp.makeConstraints { make in
+            make.leading.top.equalToSuperview()
+            make.width.equalTo(width)
+            make.height.equalTo(48)
+        }
+        
+        containerView.addSubview(countLabel)
+        countLabel.snp.makeConstraints { make in
+            make.trailing.equalToSuperview().offset(-18)
+            make.centerY.equalToSuperview()
+        }
+        
+        containerView.addSubview(nameLabel)
+        
+    }
+    
+    private func manageSelection(){
```

Comment:
[p4] type을 인자로 받아서 하면 조금 더 깔꼼하게 쓸 수 있지 않을까 하는 생각!
그러면 메서드명도 수정해서 더 잘 구분이 갈지도!
```suggestion
private func manageSelection(isSelected: Bool) {
    UIView.animate(withDuration: 0.1) {
        self.setStatus(type: isSelected ? .doneSelected : doneNotSelected )
}
```

Reply:
이렇게하면 코드가 비정상적으로 동작해서 우선 지금걸로 유지하고 데이터 바인딩하면서 수정해볼게용